### PR TITLE
feat: monitor network usage segmented by peer

### PR DESCRIPTION
**Pull request by luislhl**
design: https://github.com/HathorNetwork/ops-tools/issues/435

### Acceptance Criteria
- We want to be able to have a metric of network bandwith usage for each peer connection in the fullnodes.
- Some other metrics we were already collection will also be sent to Prometheus

### CODE REVIEW

**Review by jansegre — APPROVED** [review]

File: hathor/metrics.py [hunk 1]
```diff
@@ -254,8 +276,37 @@ def set_stratum_data(self) -> None:
         self.blocks_found = blocks_found
         self.estimated_hash_rate = estimated_hash_rate
 
+    def collect_peer_connection_metrics(self) -> None:
+        """Collect metrics for each connected peer.
+            The list is cleated every time to avoid memory leak.
```
Comment (pedroferreira1) [inline hathor/metrics.py:281]: Typo: cleared*

File: hathor/prometheus.py [hunk 2]
```diff
@@ -26,6 +26,8 @@
 
 settings = HathorSettings()
 
+METRIC_PREFIX = 'hathor_core:'
```
Comment (pedroferreira1) [inline hathor/prometheus.py:29]: Should we use the same prefix added in the other PR with the cli command?

File: hathor/prometheus.py [hunk 3]
```diff
@@ -42,6 +44,50 @@
     'send_token_timeouts': 'Number of times send_token API has timed-out',
 }
 
+# Defines the metrics related to peer connections
+PEER_CONNECTION_METRICS = {
+    "received_messages": Gauge(
+        METRIC_PREFIX + "peer_connection_received_messages",
+        "Counts how many messages the node received from a peer",
+        labelnames=["network", "connection_string", "peer_id"],
```
Comment (pedroferreira1) [inline hathor/prometheus.py:52]: Looks like all `labelnames` are equal. Should we have a variable and reuse it?

File: hathor/prometheus.py (hunk 2 above)
Comment (luislhl) [reply hathor/prometheus.py:29]: Yes, I'll do this when the other one is merged

File: hathor/prometheus.py (hunk 3 above)
Comment (luislhl) [reply hathor/prometheus.py]: Done in https://github.com/HathorNetwork/hathor-core/pull/467/commits/fbc0453ccbcbed471d13cb3f66d7481a8666b8e7

File: hathor/metrics.py (hunk 1 above)
Comment (luislhl) [reply hathor/metrics.py]: Fixed in https://github.com/HathorNetwork/hathor-core/pull/467/commits/fbc0453ccbcbed471d13cb3f66d7481a8666b8e7

File: hathor/prometheus.py (hunk 2 above)
Comment (luislhl) [reply hathor/prometheus.py:29]: I did a refactor in the merge: https://github.com/HathorNetwork/hathor-core/pull/467/commits/a16d9077d3fc33cddc97bcbfd0c3b15f7074802b

**Review by pedroferreira1 — APPROVED** [review]

File: hathor/prometheus.py (hunk 2 above)
Comment (pedroferreira1) [inline hathor/prometheus.py:29]: We are not using this anymore, right?

**Review by pedroferreira1 — APPROVED** [review]

File: hathor/prometheus.py (hunk 2 above)
Comment (luislhl) [reply hathor/prometheus.py]: Sure, I must have overseen this during merge

**Review by pedroferreira1 — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

**Pull request merged**